add launch_site event immediately after the launch button is clicked

roo2 opened the pull request:
#### Changes proposed in this Pull Request

* Added the `calypso_site_launch_start` event when the launch button is clicked after following the gutenboarding flow. This had to be done on the page after `launch` because the `launch` button is managed by the `wpcom-block-editor` app

#### Testing instructions for launch event
* [ x ] Go through the existing signup flow from `/start`
         * The `calypso_site_launch_start` event should not be triggered at any point 
* [ x ] Go through `/new` with a free domain
     * Click the `Launch` button
         * the `calypso_site_launch_start` event is triggered and you will be on the domain select page.
* [ x ] Go through `/new` with a paid domain
    * Click on the `Launch` button
         * the `calypso_site_launch_start` event is triggered and you will be on the domain select page.

ramonjd commented:
> I mean to add an event for when a site is launched (or immediately after site launch) to this PR but have been a bit stuck on that

It's not straight forward unfortunately.

Assuming we want to fire an event after a site has been launch in a callback, we'll have to dig around from the most appropriate place.

My guess is it might need some poking around the signup APIs, whose complexity is too great for this small text box :)

The launch flow is defined in our signup flows:

https://github.com/Automattic/wp-calypso/blob/master/client/signup/config/flows-pure.js#L284

This is the method that the [launch step](https://github.com/Automattic/wp-calypso/blob/master/client/signup/config/steps-pure.js#L633) fires to launch a site when the step is complete

https://github.com/Automattic/wp-calypso/blob/master//client/lib/signup/step-actions/index.js#L359

Called here:
https://github.com/Automattic/wp-calypso/blob/master/client/lib/signup/flow-controller.ts#L328

So we actually already have an event [calypso_signup_actions_complete_step](https://github.com/Automattic/wp-calypso/blob/master/client/lib/signup/flow-controller.ts#L335) that is firing for the launch step it seems. The `flow` is `launch-site` and the `step` is `launch`.

I wonder if that's enough? 

Unless we want to know if the **site** in question has been created via Gutenboarding?

roo2 commented:
Hmm I expected to see that `calypso_signup_actions_complete_step` event fired on the next page immediately after the launch button was clicked. but it wasn't triggered yet. @johnHackworth is that where you wanted the event to be triggered?

I've implemented a new `calypso_site_launch_start` event using the following implementation/hack:

When the user clicks the launch button, they're taken to `start/new-launch` code in https://github.com/Automattic/wp-calypso/blob/master/client/signup/controller.js#L172 then redirects the user to the start of the launch flow which is`/start/new-launch/domains-launch`.

I've added an extra condition to https://github.com/Automattic/wp-calypso/blob/master/client/signup/controller.js that will record a `calypso_site_launch_start` event when we hit the `/start/new-launch` domain.

Then the redirection to `/start/new-launch/domains-launch` continues as normal. This is good because if the user refreshes the page for example, they won't get two launch events.

ramonjd commented:
Thanks for persevering with this one.

It works as described.

I'm wondering if it fulfils what we're after. (Original discussion p1589148343104200-slack-CNMU1UHHB)

`wpcom_site_launched` is recorded in the backend **after** the launch status blog option is set to `launched`

`calypso_site_launch_start` on the other hand is fired when I click the launch button, and even if I decide to abandon the launch flow.

Do we want to record when a site is launched or when the user decides to enter the launch flow?

ramonjd commented:
> I'm wondering if it fulfils what we're after. 

If the _"where to put launch tracking"_ question is holding the domains part up, we should address it in another PR

roo2 commented:
> > I'm wondering if it fulfils what we're after.
> 
> If the _"where to put launch tracking"_ question is holding the domains part up, we should address it in another PR

Yea too true, I'll break them out now

roo2 commented:
created https://github.com/Automattic/wp-calypso/pull/42245

razvanpapadopol commented on client/signup/controller.js:
```diff
@@ -144,6 +144,13 @@ export default {
+		// We record the event here because the launch button itself is managed by the wpcom-block-editor app
+		// wpcom-block-editor app may be treated differently by the ad blocker but this code is guaranteed to be hit.
+		if ( context.pathname === '/start/new-launch' ) {
+			recordTracksEvent( 'calypso_site_launch_start' );
```
Since we're firing this event only for `/new-launch` shouldn't we name it `calypso_newsite_*` event or add `flow: 'gutenboarding'` prop?

simison replied:
`calypso_newsite_launch_start` seems fine to me 👍

simison commented on client/signup/controller.js:
```diff
@@ -144,6 +144,13 @@ export default {
 
+		// Record a `calypso_*` tracks event after clicking the launch button.
+		// We record the event here because the launch button itself is managed by the wpcom-block-editor app
+		// wpcom-block-editor app may be treated differently by the ad blocker but this code is guaranteed to be hit.
```
I don't understand this bit — why would it be different? Anyway, we'll move the launch button to FSE plugin I think and we should be able to fire events from there too.
